.github/workflows/build-distributables.yml

```diff
@@ -29,26 +33,145 @@ jobs:
     - uses: conda-incubator/setup-miniconda@v3
       with:
         auto-update-conda: true
-        python-version: ${{ matrix.config.py }}
+        python-version: ${{ env.PYTHON_VERSION }}
+        miniforge-version: latest
+        activate-environment: builder
 
     - name: Build conda-packed (all platforms)
-      shell: bash -el {0}
       run: |
+        conda install -y versioningit nodejs
         conda activate base
         ./extra/build_conda_packed.sh
-        ls artifacts/*
-        {
-          echo 'DISTRIBUTABLES<<EOF'
-          ls artifacts/*
-          echo EOF
-        } >> "$GITHUB_ENV"
+
+    - name: Create MacOS App
+      if: startsWith(matrix.config.os, 'macos')
+      env:
+          MACOS_CERTIFICATE: ${{ secrets.MACOS_CERTIFICATE_ISA }}
+          MACOS_CERTIFICATE_PWD: ${{ secrets.MACOS_CERTIFICATE_ISA_PWD }}
+          KEYCHAIN_PASSWORD: ${{ secrets.KEYCHAIN_PASSWORD }}
+          NOTARIZATION_USERNAME: ${{ secrets.MACOS_NOTARIZATION_USERNAME }}
+          NOTARIZATION_PASSWORD: ${{ secrets.MACOS_NOTARIZATION_PASSWORD }}
+          DEVELOPER_IDENTITY: "Developer ID Application: The International Scattering Alliance (8CX8K63BQM)"
+      run: |
+        echo $MACOS_CERTIFICATE | base64 --decode > certificate.p12
+        security create-keychain -p $KEYCHAIN_PASSWORD build.keychain
+        security default-keychain -s build.keychain
+        security unlock-keychain -p $KEYCHAIN_PASSWORD build.keychain
+        security import certificate.p12 -k build.keychain -P $MACOS_CERTIFICATE_PWD -T /usr/bin/codesign
+        security set-key-partition-list -S apple-tool:,apple:,codesign: -s -k $KEYCHAIN_PASSWORD build.keychain
+        security find-identity -p codesigning
+
+        echo "Creating MacOS App"
+        # This assumes that conda-packed enviroment exists at "conda_packed"
+
+        VERSION=$(versioningit)
+        PRODUCT_NAME="$PACKAGE_NAME-$VERSION"
+        FULL_PACKAGE_NAME="$PRODUCT_NAME-$(uname -s)-$(uname -m)"
+        BUILD_DIR="${PACKAGE_NAME}_installer"
+        mkdir -p $BUILD_DIR
+        APP_FOLDER="$BUILD_DIR/$PRODUCT_NAME"
+        APP="$APP_FOLDER/${PACKAGE_NAME}.app"
+        PYTHON_FRAMEWORK="$APP/Contents/Frameworks/python.framework"
+        PYTHON_ENV="$PYTHON_FRAMEWORK/Resources/env"
+        ln -s /Applications "$BUILD_DIR/Applications"
+        cp -R -P "extra/platform_resources/macos/app_folder_template" "$APP_FOLDER"
+        cp -R -P "conda_packed" "$PYTHON_ENV"
+        plutil -replace CFBundleVersion -string "$VERSION" "$APP/Contents/Info.plist"
```

Comment:
Do we also need to set CFBundleShortVersionString which is the user visible property?

https://developer.apple.com/documentation/bundleresources/information-property-list/cfbundleshortversionstring

Also, there appears to be an internal python framework. Should this be tagged with the python version number used?

What about the refl1d_shell app Info.plist?

Reply:
The refl1d_shell.app is just a script that opens a shell with the environment from refl1d.app active - the shell app itself has no meaningful version.

It is only meaningful to put versions on the framework if we are planning to upgrade it or use it independently of the app.